libs/sdk-ui-dashboard/src/model/state/_infra/queryCall.ts

```diff
@@ -0,0 +1,53 @@
+// (C) 2021 GoodData Corporation
+
+import { IDashboardQuery } from "../../queries";
+import noop from "lodash/noop";
+import { QueryEnvelope, QueryEnvelopeActionTypeName } from "./queryProcessing";
+import { call, put } from "redux-saga/effects";
+import { SagaIterator } from "redux-saga";
+
+function envelopeWithPromise<TResult>(q: IDashboardQuery<TResult>): {
+    promise: Promise<TResult>;
+    envelope: QueryEnvelope;
+} {
+    const partialEnvelope = {
+        onStart: noop,
```

Comment:
This seems to never be set, or am I missing something?

Reply:
nope, it's intentionally noop in this context. while using query generator, caller is not interested in when it started, just the end result.

it will be useful in the `useDashboardQuery` in presentational components that may want to distinguish the start state